routes/health.js

```diff
@@ -0,0 +1,54 @@
+var system = require('../system');
+
+var mongoose = require('mongoose');
+
+const Errors = {
+    DBERROR: 'DBERROR'
+}
+
+exports.gethealth = function (req, res) {
+    const isHealthEndpointEnabled = system.isHealthEndpointEnabled();
+    if (!isHealthEndpointEnabled) {
+        return res.status(404).send("not found");
+    } else {
+        const mongoose_state = mongoose.connection.readyState
+        var errors = collectErrors();
+        if (errors.length == 0) {
+            return res.status(200).json({
+                status: "OK",
+                mongoose: mongoose_state
+            });
+        } else {
+            return res.status(500).json({
+                status: "Not OK",
+                mongoose: mongoose_state,
+                errors: errors
+            });
+        }
+
+        function collectErrors() {
+            var errors = [];
```

Comment:
Curious why this returns an array, it looks like to me this can ever have one value?   Why not return the error itself ?

Reply:
At the moment the health endpoint just checks for mongodb connection errors. I thought of adding more checks later, for example for redis or some of the integration like ifttt.

Reply:
Got it, redis sounds like a good candidate.